libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -2918,10 +2918,7 @@ void IRGeneratorForStatements::setLValue(Expression const& _expression, IRLValue
 	solAssert(!m_currentLValue, "");
 
 	if (_expression.annotation().willBeWrittenTo)
-	{
 		m_currentLValue.emplace(std::move(_lvalue));
-		solAssert(!_lvalue.type.dataStoredIn(DataLocation::CallData), "");
```

Comment:
Acutally can you replace this by:
If the type is stored in calldata, then the _lvalue.kind is stack?

Reply:
Added @chriseth 
https://github.com/ethereum/solidity/pull/10912